##  Value at Risk (VaR) — ONGC (Nov 2025 – Apr 2026)

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import yfinance as yf

data = yf.download('ONGC.NS', start='2025-11-01', end='2026-04-30', progress=False)['Close'].squeeze()

returns = data.pct_change().dropna()

var_90 = returns.quantile(0.10)
var_95 = returns.quantile(0.05)
var_99 = returns.quantile(0.01)

print(f"VaR 90%: {var_90:.2%}")
print(f"VaR 95%: {var_95:.2%}")
print(f"VaR 99%: {var_99:.2%}")

VaR 90%: -1.46%
VaR 95%: -2.05%
VaR 99%: -3.23%


After running Monte Carlo to see possible future prices, I wanted 
to answer a more specific question — on a really bad day, how much 
can I actually lose on ONGC? That's exactly what Value at Risk tells you.

I calculated VaR at three confidence levels during the peak of the 
Iran-USA geopolitical crisis period:

| Confidence Level | VaR | Meaning |
|-----------------|-----|---------|
| 90% | -1.46% | 1 in 10 days could be worse than this |
| 95% | -2.05% | 1 in 20 days could be worse than this |
| 99% | -3.23% | 1 in 100 days could be worse than this |

---

### In real money terms on INR 1,00,000 investment:
VaR 90% → on 90 out of 100 days,
you won't lose more than INR 1,460
VaR 95% → on 95 out of 100 days,
you won't lose more than INR 2,050
VaR 99% → on 99 out of 100 days,
you won't lose more than INR 3,230

---

### What this means during the geopolitical crisis:

Honestly these numbers are not as scary as I expected for a stock 
going through such a turbulent period. A 2% daily loss limit at 
95% confidence is actually pretty reasonable for an oil stock 
during active geopolitical tensions.

But here's the catch — VaR only tells you about the 95% or 99% 
of days. It says nothing about what happens on that 1 bad day 
out of 100. In the geopolitical crisis we saw ONGC drop much 
more than 3.23% on single days when major escalation news hit. 
That's the limitation of VaR.

---

### Connecting VaR to Monte Carlo:

Remember from Monte Carlo — worst case for ONGC was INR 143 
from current INR 284. That's a 50% drop over a year.

VaR now tells us that on any single day during this period, 
99% of the time the loss won't exceed 3.23%. 

So the 50% annual drop from Monte Carlo doesn't happen in 
one day — it's a slow painful series of -2% and -3% days 
adding up over time. That's actually a really important 
insight for position sizing in algo trading.

---

### Key Takeaway for Trading 

VaR is used by every serious trading desk in the world to 
answer one question — "how much can I lose today in the 
worst case?" 

For ONGC during the Iran-USA crisis:
- Safe to hold with a stop loss around 2-3% per day
- Anything beyond 3.23% daily loss is a tail risk event
- During geopolitical news days always expect tail risk
  to kick in — those are the 1 in 100 days VaR doesn't protect you from